### PR TITLE
Using Memcpy to make the object

### PR DESCRIPTION
**Pull request by basimsahaf**
This PR is to review the memcpy way of constructing the object. Please refer to https://github.com/googleinterns/scsi2nvme/pull/20/files#r427565925 to review the bit_cast way of constructing the object. Would love to know your thoughts.

### CODE REVIEW

File: test/scsi_defs_test.cc [hunk 1]
```diff
@@ -32,4 +34,34 @@ namespace {
...
+    (evpd << 8) +
+    (obsolete << 9) +
+    (reserved << 10) +
+    (pageCode << 16) + 
+    (allocLength << 24) +
+    (control << 40));
+
+    // Two ways of getting the struct. 1. Using memcpy:
+    scsi_defs::InquiryCommand ict;
+    memcpy(&ict, &cmdValue, sizeof(scsi_defs::InquiryCommand));
```
Comment (basimsahaf) [inline test/scsi_defs_test.cc:57]: memcpy version of constructing the object

File: lib/scsi_defs.h [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    OpCode opCode = OpCode::kInquiry;
+    bool evpd: 1; // Enable Vital Product Data (EVPD)
+    bool obsolete: 1; // formerly CMDDT
+    uint8_t reserved: 6;
+    uint8_t pageCode: 8;
+    uint16_t allocationLength: 16;
+    ControlByte controlByte;
+    // uint16_t padding: 16; // Don't need padding for memcpy
+
+  } ABSL_ATTRIBUTE_PACKED;
```
Comment (w-michal) [inline lib/scsi_defs.h:131]: Add `static_assert(sizeof(InquiryCommand) == ` with the expected size of the struct.

File: lib/scsi_defs.h [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+
+  // Refer to https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf , Section 3.6 Table 58
+  struct InquiryCommand {
+    OpCode opCode = OpCode::kInquiry;
+    bool evpd: 1; // Enable Vital Product Data (EVPD)
+    bool obsolete: 1; // formerly CMDDT
+    uint8_t reserved: 6;
+    uint8_t pageCode: 8;
+    uint16_t allocationLength: 16;
```
Comment (w-michal) [inline lib/scsi_defs.h:127]: Add comment about endianness of this field (is it big or little endian?).

**Comment by basimsahaf** [comment]
closing this PR as we're no longer using memcpy

File: lib/scsi_defs.h (hunk 2 above)
Comment (basimsahaf) [reply lib/scsi_defs.h:131]: good idea!

**Pull request closed without merging**